packages/yswaps/scripts/libraries/solvers/multicall/CurveYfiEth.ts

```diff
@@ -52,81 +64,115 @@ export class CurveYfiEth implements Solver {
+
+    console.log('[CurveYfiEth] CRV management');
+    const shouldExecuteCrvTrade = crvBalance.gt(DUST_THRESHOLD);
+
+    if (shouldExecuteCrvTrade) {
+      console.log('[CurveYfiEth] Transfering crv to multicall swapper for simulations');
+      await this._crv.connect(strategySigner).transfer(this._multicallSigner._address, crvBalance);
+
+      // Do gas optimizations
+      console.log('[CurveYfiEth] Getting crv => weth trade information via zrx');
+      const { data: zrxCrvData, allowanceTarget: zrxCrvAllowanceTarget } = await zrx.quote({
+        chainId: 1,
+        sellToken: this._crv.address,
+        buyToken: this._weth.address,
+        sellAmount: crvBalance,
+        slippagePercentage: 10 / 100,
+      });
+
+      const approveCrv = (await this._crv.allowance(this._multicallSigner._address, zrxCrvAllowanceTarget)).lt(crvBalance);
+      if (approveCrv) {
+        console.log('[CurveYfiEth] Approving crv');
+        const approveCrvTx = await this._crv.populateTransaction.approve(zrxCrvAllowanceTarget, constants.MaxUint256);
+        await this._multicallSigner.sendTransaction(approveCrvTx);
+        transactions.push(approveCrvTx);
+      }
+
+      console.log('[CurveYfiEth] Executing crv => weth via zrx');
+      const crvToWethTx = {
+        to: this.zrxContractAddress,
+        data: zrxCrvData,
+      };
+      await this._multicallSigner.sendTransaction(crvToWethTx);
+      transactions.push(crvToWethTx);
+
+      asyncTradesExecutionDetails.push({
+        _strategy: strategy,
+        _tokenIn: this._crv.address,
+        _tokenOut: this._crvYfiEth.address,
+        _amount: crvBalance,
+        _minAmountOut: BigNumber.from('0'), // populated later when minAmountOut calculated.
+      });
+    } else {
+      console.log('[CurveYfiEth] Should NOT execute CRV trade');
     }
 
-    console.log('[CurveYfiEth] Executing crv => weth via zrx');
-    const crvToWethTx = {
-      to: this.zrxContractAddress,
-      data: zrxCrvData,
-    };
-    await multicallSwapperSigner.sendTransaction(crvToWethTx);
-    transactions.push(crvToWethTx);
-
-    console.log('[CurveYfiEth] Getting cvx => weth trade information via zrx');
-    const { data: zrxCvxData, allowanceTarget: zrxCvxAllowanceTarget } = await zrx.quote({
-      chainId: 1,
-      sellToken: this._cvx.address,
-      buyToken: this._weth.address,
-      sellAmount: cvxBalance,
-      slippagePercentage: 10 / 100,
-    });
-
-    const approveCvx = (await this._cvx.allowance(multicallSwapperAddress, zrxCvxAllowanceTarget)).lt(cvxBalance);
-    if (approveCvx) {
-      console.log('[CurveYfiEth] Approving cvx');
-      const approveCvxTx = await this._cvx.populateTransaction.approve(zrxCvxAllowanceTarget, constants.MaxUint256);
-      await multicallSwapperSigner.sendTransaction(approveCvxTx);
-      transactions.push(approveCvxTx);
+    console.log('[CurveYfiEth] CVX management');
+    const shouldExecuteCvxTrade = cvxBalance.gt(DUST_THRESHOLD);
```

Comment:
I don't know if it makes sense to execute these trades separately.
Perhaps something like:
```
cvxBalance.gt(DUST_THRESHOLD) || crvBalance.gt(DUST_THRESHOLD)
```

makes more sense

Reply:
they should increase together since it's a convex strat.